integration/setup/setup.go

```diff
@@ -92,3 +105,50 @@ func Resources(c *client.AzureClientSet, g *framework.Guest, h *framework.Host)
 
 	return nil
 }
+
+func installCredential(h *framework.Host) error {
+	o := func() error {
+		k8sClient := h.K8sClient()
+
+		k8sClient.CoreV1().Secrets(credentialNamespace).Delete(credentialName, &metav1.DeleteOptions{})
+
+		c := client.AzureClientSetConfig{
+			ClientID:       env.AzureGuestClientID(),
+			ClientSecret:   env.AzureGuestClientSecret(),
+			SubscriptionID: env.AzureGuestSubscriptionID(),
+			TenantID:       env.AzureGuestTenantID(),
+		}
```

Comment:
I just don't follow why should we use `AzureClientSetConfig` here.

You could just do:

```
+				credential.ClientIDKey:       []byte(env.AzureGuestClientID()),
+				credential.ClientSecretKey:   []byte(env.AzureGuestClientSecret()),
+				credential.SubscriptionIDKey: []byte(env.AzureGuestSubscriptionID()),
+				credential.TenantIDKey:       []byte(env.AzureGuestTenantID()),
```

But being in scope of a single method it's now easier to realise that `AzureClientSetConfig` isn't use to create actual client, so it isn't a blocker.

Reply:
you are right, no need for that. removing this